src/app/shared/components/protected-component/protected.pipe.ts

```diff
@@ -0,0 +1,17 @@
+import { Pipe, PipeTransform } from '@angular/core';
+import { ProfileService } from '../../../core/profile/profile.service';
+
+@Pipe({ name: 'checkLogin', pure: false })
+export class CheckLoginPipe implements PipeTransform {
```

Comment:
As for me this is wrong pipe use case, you can do the same with the controller.
I would avoid pipes that transform values to boolean